administrator/components/com_fields/models/field.php

```diff
@@ -590,7 +590,7 @@ public function setFieldValue($fieldId, $itemId, $value)
 		elseif (count($value) == 1 && count((array) $oldValue) == 1)
 		{
 			// Only a single row value update can be done when not empty
-			$needsUpdate = is_array($value[0]) ? !count($value[0]) : !strlen($value[0]);
+			$needsUpdate = is_array($value[0]) ? count($value[0]) === 0 ? false : true : strlen($value[0]) === 0 ? false : true;
```

Comment:
What I mean is, does the following code the same?
`$needsUpdate = is_array($value[0]) ? count($value[0]) !== 0 : strlen($value[0]) !== 0;`

Reply:
Yes, it works and shorter/more readable.

Reply:
top, works! Way better then my fiddle :)